adapters/zio-http/src/main/scala/caliban/ZHTTPAdapter.scala

```diff
@@ -0,0 +1,239 @@
+package caliban
+
+import zio._
+import zio.clock.Clock
+import zio.duration._
+import zio.stream._
+
+import caliban.{ GraphQLInterpreter, GraphQLRequest, GraphQLResponse }
+import caliban.ResponseValue.{ ObjectValue, StreamValue }
+import caliban.ResponseValue
+import caliban.Value.NullValue
+import caliban.execution.QueryExecution
+import io.circe._
+import io.circe.parser._
+import io.circe.syntax._
+import zhttp.http._
+import zhttp.service._
+import zhttp.socket.SocketApp
+import zhttp.socket.WebSocketFrame.Text
+import zhttp.socket._
+
+object ZHttpAdapter {
+  case class GraphQLWSRequest(`type`: String, id: Option[String], payload: Option[GraphQLRequest])
```

Comment:
Should we move this to somewhere else since `graphql-ws` is what all adapters support?

Reply:
I think that's ok, there is no "common" project and we can't put it in core because it depends on circe.